.github/workflows/step_tests-pip.yml

```diff
@@ -0,0 +1,77 @@
+name: test-pip
+run-name: Run tests using Pip
+
+on:
+  workflow_call:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test-pip:
+    continue-on-error: ${{ matrix.experimental || false }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ 3.8, 3.9, "3.10", "3.11"]
+        markdown-it-py-version: ["~=2.0"]
+        kernel: [true]
+        include:
+          - python-version: "3.12-dev"
+            experimental: true
+          - python-version: "3.13-dev"
+            experimental: true
+          - python-version: 3.9
+            markdown-it-py-version: ""
+          - python-version: 3.9
+            markdown-it-py-version: "~=3.0"
+          - python-version: "3.11"
+            markdown-it-py-version: "~=4.0"
+            experimental: true
+          - python-version: 3.9
+            kernel: false
+          - python-version: 3.9
+            quarto: true
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+        with:
+          python_version: ${{ matrix.python-version }}
+
+      - name: Upgrade pip
+        run: |
+          python -m pip install --upgrade pip
+
+      - name: Install markdown-it-py
+        if: ${{ matrix.markdown-it-py-version }}
+        run: python -m pip install markdown-it-py${{ matrix.markdown-it-py-version }}
+
+      - name: Install from source (required for the pre-commit tests)
+        run: python -m pip install '.[test-cov]'
```

Comment:
I dont think we can merge them because there is a case where we want to test `jupytext` without `markdown-it-py-version`. If we merge them, even `jupytext` installation will be skipped as well.

Reply:
The following works fine because `markdown-it-py` is a hard dependency so it will always be installed
```yaml
run: python -m pip install -e .[test-cov] markdown-it-py${{ matrix.markdown-it-py-version }}
```
or, if we want to really not specify the latter
```yaml
run: python -m pip install -e .[test-cov] ${{ matrix.markdown-it-py-version && format('markdown-it-py{0}', matrix.markdown-it-py-version }}
```
Let me know if the latter doesn't work